src/data/reducers/auth.js

```diff
@@ -24,11 +24,21 @@ export default function (state = initialState, action) {
       return {
         ...state,
         isLoggedIn: false,
+        firstName: null,
+        lastName: null, 
+        email: null,
+        userId: null,
+        teamId: null,
       };
```

Comment:
All the actions besides `LOGIN_SUCCESS` seem to return the same thing - not sure if there was a reason for this. 

I also think it'll be easier to maintain state across actions if `initialState` isn't as loosely defined. The current implementation spreads state and then sets all the properties to null making it repetitive. Alternatively, if `initialState` is strictly defined, we could spread the initial state and update only the fields relating to the action.

```suggestion
      return {
        ...initialState,
       propertyRelatedToAction: payload.asdasd
      };
```

I know it's not in your ticket and was implemented this way before. I'm just putting the idea out there to get your thoughts and @jzxhuang because I think this pattern won't be maintainable for more complex reducers.

Reply:
Yeah that's a really good point, maybe the previous devs anticipated that there would be different payloads for the actions? I'm not too sure what the purpose is, but we can definitely keep this in mind as we're adding more to the state